system-tests/src/test/java/org/eclipse/edc/samples/transfer/HttpRequestLoggerUtil.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.samples.transfer;
+
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.output.ToStringConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import static org.eclipse.edc.samples.transfer.FileTransferCommon.getFileFromRelativePath;
+
+public class HttpRequestLoggerUtil {
+
+    private static final String HTTP_REQUEST_LOGGER_DOCKER_COMPOSE_FILE_PATH = "util/http-request-logger/docker-compose.yaml";
+
+    private static final String HTTP_REQUEST_LOGGER = "http-request-logger";
+
+    public static DockerComposeContainer<?> getHttpRequestLoggerContainer() {
+        return new DockerComposeContainer<>(getFileFromRelativePath(HTTP_REQUEST_LOGGER_DOCKER_COMPOSE_FILE_PATH))
+                .withLocalCompose(true)
+                .waitingFor(HttpRequestLoggerUtil.HTTP_REQUEST_LOGGER, Wait.forLogMessage(".*started.*", 1));
+    }
+
+    public static DockerComposeContainer<?> getHttpRequestLoggerContainer(ToStringConsumer toStringConsumer) {
+        return getHttpRequestLoggerContainer()
+                .withLogConsumer(HTTP_REQUEST_LOGGER, toStringConsumer);
+    }
```

Comment:
I think I had trouble specifying the context for the testcontainers
I use the docker compose file as a wrapper to start the container with the needed root context
If you have any idea how I can work around this without docker compose I'm totally on your side

Reply:
I've moved the docker compose file (http-request-logger-wrapper.yaml) to the test directory since it's only relevant for testing

The guide does not refer to the compose file anymore

Reply:
The root context is not working because you are generally building the http-request-logger docker image specifying the dockerfile path (see in the prerequisite readme: `docker build -f util/http-request-logger/Dockerfile -t http-request-logger .`) and not the context path.

The correct build command should be:
```
docker build -t http-request-logger util/http-request-logger
```

doing this, in the dockerfile the RUN command will be:
```
RUN gradle build
```

and the COPY will be:
```
COPY --from=build /home/gradle/project/build/libs/http-request-logger.jar /app/http-request-logger.jar
```

at that point, in the tests you can use a `GenericContainer` (my advice would be to extend it) that will get as the constructor parameter something like this:
```java
new ImageFromDockerfile().withDockerfile(getFileFromRelativePath(HTTP_REQUEST_LOGGER_DOCKERFILE_FILE_PATH).toPath())
```